packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -251,15 +251,19 @@ DataGrid.propTypes = {
   selectionModel: chainPropTypes(
     PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.array]),
     (props: any) => {
-      if (Array.isArray(props.selectionModel) && props.selectionModel.length > 1) {
+      if (
+        props.checkboxSelection === false &&
```

Comment:
Initially this prop is `undefined`.

```suggestion
        !props.checkboxSelection &&
```

Reply:
We mention default as `false`:
https://github.com/mui-org/material-ui-x/blob/f1538f0cc386c147a5919e540773b1deecff3d35/packages/grid/_modules_/grid/models/gridOptions.tsx#L50-L53

and also in docs.

I think it would be better if we keep this strict check and instead add `checkboxSelection: false` in `DEFAULT_GRID_OPTIONS`. What do you think?

Reply:
Interestingly, even if I add `checkboxSelection: false` in `DEFAULT_GRID_OPTIONS`, I get checkboxSelection prop as `undefined` initially in tests.